server/actions_artifacts.go

```diff
@@ -96,7 +96,10 @@ func (s *RegistryServer) CreateArtifact(ctx context.Context, req *rpc.CreateArti
 		}
 	}
 
-	artifact := models.NewArtifact(name, req.GetArtifact())
+	artifact, err := models.NewArtifact(name, req.GetArtifact())
+	if err != nil {
+		return nil, err
```

Comment:
This is a good pattern - since most of the other calls in this file are returning already-wrapped errors, I put the error-wrapping inside models.NewArtifact (where the gzipping might fail). I think we should do the same for models.NewSpec